src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (!(target instanceof Herbivore) || target.hidden) {
+      return; // Bite doesn't work on carnivores or hidden herbivores
+    }
+
+    target.health -= 50;
+
+    if (target.health <= 0) {
+      target.die();
+    }
```

Comment:
Before calling `target.die()`, ensure that the target is not already dead. This can be done by checking if the target is still in the `Animal.alive` array.